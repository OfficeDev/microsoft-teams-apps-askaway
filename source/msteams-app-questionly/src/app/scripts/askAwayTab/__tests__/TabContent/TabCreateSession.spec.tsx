/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, Button, Image } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import TabCreateSession from '../../TabContent/TabCreateSession';

configure({ adapter: new Adapter() });

describe('TabCreateSession Component', () => {
    const t = jest.fn();
    const showTaskModule = jest.fn();
    it('should match the snapshot', () => {
        const wrapper = shallow(<TabCreateSession t={t} showTaskModule={showTaskModule} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<TabCreateSession t={t} showTaskModule={showTaskModule} />);

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Image)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(2);
        expect(component.find(Button)).toHaveLength(1);
    });
});
