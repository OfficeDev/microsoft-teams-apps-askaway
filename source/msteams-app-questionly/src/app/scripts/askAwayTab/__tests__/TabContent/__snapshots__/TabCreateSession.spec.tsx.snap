// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabCreateSession Component should match the snapshot 1`] = `
<Flex
  className="screen"
  hAlign="center"
  vAlign="center"
>
  <Image
    accessibility={[Function]}
    alt="image"
    as="img"
    className="create-session"
    src="collaboration.png"
  />
  <FlexItem
    align="center"
  >
    <Text
      as="span"
      className="text-caption"
    />
  </FlexItem>
  <FlexItem
    align="center"
  >
    <Text
      as="span"
      className="text-subcaption"
    />
  </FlexItem>
  <FlexItem
    align="center"
  >
    <Button
      accessibility={[Function]}
      as="button"
      className="button"
      onClick={[Function]}
      primary={true}
      size="medium"
    >
      <ButtonContent
        as="span"
      />
    </Button>
  </FlexItem>
</Flex>
`;
